sql/src/main/codegen/includes/replace.ftl

```diff
@@ -20,33 +20,52 @@
 // Taken from syntax of SqlInsert statement from calcite parser, edited for replace syntax
 SqlNode DruidSqlReplaceEof() :
 {
-    SqlNode table;
+    final SqlIdentifier destination;
     SqlNode source;
     SqlNodeList columnList = null;
     final Span s;
+    SqlNode tableRef = null;
     SqlInsert sqlInsert;
     // Using fully qualified name for Pair class, since Calcite also has a same class name being used in the Parser.jj
     org.apache.druid.java.util.common.Pair<Granularity, String> partitionedBy = new org.apache.druid.java.util.common.Pair(null, null);
     SqlNodeList clusteredBy = null;
     final Pair<SqlNodeList, SqlNodeList> p;
     SqlNode replaceTimeQuery = null;
+    String exportFileFormat = null;
 }
 {
     <REPLACE> { s = span(); }
     <INTO>
-    table = CompoundIdentifier()
-    [
-        p = ParenthesizedCompoundIdentifierList() {
-            if (p.left.size() > 0) {
-                columnList = p.left;
-            }
+    (
+      LOOKAHEAD(2)
+      <EXTERN> <LPAREN> destination = ExternalDestination() <RPAREN>
+      |
+      destination = CompoundTableIdentifier()
+      ( tableRef = TableHints(destination) | { tableRef = destination; } )
+      [ LOOKAHEAD(5) tableRef = ExtendTable(tableRef) ]
+    )
+    (
+      LOOKAHEAD(2)
+      p = ParenthesizedCompoundIdentifierList() {
+        if (p.right.size() > 0) {
+          tableRef = extend(tableRef, p.right);
         }
+        if (p.left.size() > 0) {
+          columnList = p.left;
+        } else {
+          columnList = null;
+        }
+      }
+      | { columnList = null; }
```

Comment:
Seems like there is some code duplication b/w INSERT and REPLACE. Can we unify some code, and only keep the high-level constructs in place?